### PR TITLE
chore: fix focusing dots and semicolons

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
This PR adds linter checks to ensure that
* focusing dots are centred (·) and not isolated (mathport may sometimes produce examples of the second kind)
* there are no instances of a space _before_ a semicolon, the space being unnecessary here.

It also removes some redundant semicolons.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/lint-style.py [hunk 1]
```diff
@@ -283,13 +289,17 @@ def format_errors(errors):
             output_message(path, line_nr, "ERR_TAC", "Files in mathlib cannot import the whole tactic folder")
         if errno == ERR_IBY:
             output_message(path, line_nr, "ERR_IBY", "Line is an isolated 'by'")
+        if errno == ERR_DOT:
+            output_message(path, line_nr, "ERR_DOT", "Line is an isolated focusing dot or uses . instead of · for it")
```
Comment (semorrison) [inline scripts/lint-style.py:293]:
```suggestion
            output_message(path, line_nr, "ERR_DOT", "Line is an isolated focusing dot or uses . instead of ·")
```

File: test/Monotonicity.lean [hunk 2]
```diff
@@ -137,10 +137,10 @@ example {x y z w : ℕ} : true := by
...
 --   revert ys zs
 --   induction' xs with x xs
---   ; intros ys zs h h'
---   ; cases ys with y ys
---   ; cases zs with z zs
---   ; try { cases h ; cases h' ; done },
+--   intros ys zs h h'
+--   cases ys with y ys
+--   cases zs with z zs
+--   try { cases h; cases h'; done },
```
Comment (semorrison) [inline test/Monotonicity.lean:143]: I appreciate this is commented out, but removing these semicolons would change meaning, no?

Comment (Parcly-Taxel) [reply test/Monotonicity.lean]: This piece of code has not been fully converted to Lean 4 syntax. [The original in mathlib](https://github.com/leanprover-community/mathlib/blob/728ef9dbb281241906f25cbeb30f90d83e0bb451/test/monotonicity.lean#L115C15-L115C15) also uses semicolons, which becomes `<;>` in Lean 4; I've updated the syntax (but not the names).

File: Mathlib/Data/Polynomial/Degree/Lemmas.lean [hunk 3]
```diff
@@ -131,8 +130,8 @@ theorem natDegree_mul_C_eq_of_mul_eq_one {ai : R} (au : a * ai = 1) :
 set_option linter.uppercaseLean3 false in
 #align polynomial.nat_degree_mul_C_eq_of_mul_eq_one Polynomial.natDegree_mul_C_eq_of_mul_eq_one
 
-/-- Although not explicitly stated, the assumptions of lemma `nat_degree_mul_C_eq_of_mul_ne_zero`
-force the polynomial `p` to be non-zero, via `p.leading_coeff ≠ 0`.
+/-- Although not explicitly stated, the assumptions of lemma `natDegree_mul_C_eq_of_mul_ne_zero`
+force the polynomial `p` to be non-zero, via `p.leadingCoeff ≠ 0`.
```
Comment (jcommelin) [inline Mathlib/Data/Polynomial/Degree/Lemmas.lean:134]: This PR is starting to do too many fixes at once. That makes it really hard to review. The diff is massive, so it should be extremely unsurprising. Please make changes like this in a different PR.

File: Mathlib/RingTheory/Multiplicity.lean [hunk 4]
```diff
@@ -481,31 +481,24 @@ theorem finite_mul_aux {p : α} (hp : Prime p) {a b : α} :
...
-        finite_mul_aux hp ha hpx
-        ⟨s, mul_right_cancel₀ hp.1 (by
-              rw [add_assoc, tsub_add_cancel_of_le (succ_le_of_lt hm0)]
-              simp_all [mul_comm, mul_assoc, mul_left_comm, pow_add])⟩
+          hb (hx.symm ▸ ⟨y, mul_right_cancel₀ hp.1 <| by
+            rw [tsub_add_cancel_of_le (succ_le_of_lt hm0)] at hy
+            simp [hy, pow_add, mul_comm, mul_assoc, mul_left_comm]⟩)
+        finite_mul_aux hp ha hpx ⟨s, mul_right_cancel₀ hp.1 (by
+          rw [add_assoc, tsub_add_cancel_of_le (succ_le_of_lt hm0)]
+          simp_all [mul_comm, mul_assoc, mul_left_comm, pow_add])⟩
```
Comment (jcommelin) [inline Mathlib/RingTheory/Multiplicity.lean:501]: Idem dito for this kind of reformatting and reflowing of proofs.

Ideally a diff that touches > 30 files will be completely automatically generated by a script that fixes the syntax. Then we can have high confidence that the diff *does one and does it well*.

File: Mathlib/Data/Polynomial/Degree/Lemmas.lean (hunk 3 above)
Comment (Parcly-Taxel) [reply Mathlib/Data/Polynomial/Degree/Lemmas.lean:134]: Please review #5708.
